spec/acceptance/dataset_items_spec.rb

```diff
@@ -0,0 +1,481 @@
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { no_access_token }
+    standard_request_options(:post, 'FILTER (as no access)', :ok,
+                             {response_body_content: ['200'], expected_json_path: 'data', data_item_count: 0})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:post, 'FILTER (as invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  # not logged in users can filter dataset items, but they won't get any items that they don't have permission for
+  post '/dataset_items/filter' do
+    standard_request_options(:post, 'FILTER (as not logged in)', :ok,
+                             {response_body_content: ['200'], expected_json_path: 'data', data_item_count: 0})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    let(:raw_post) { {
+      'filter' => {
+          'start_time_seconds' => {
+              'in' => ['11', '7', '100', '4']
+          }
+      },
+      'projection' => {
+          'include' => ['id', 'start_time_seconds', 'audio_recording_id', 'creator_id']
+      }
+    }.to_json }
+    standard_request_options(:post, 'FILTER (as reader)', :ok,
+                             {
+                                 expected_json_path: 'data/0/start_time_seconds',
+                                 data_item_count: 1,
+                                 regex_match: /"in"\:\[\"11\",\"7\",\"100\",\"[0-9]+\"\]/,
+                                 response_body_content: "\"start_time_seconds\":",
+                                 invalid_content: "end_time_seconds"
+                             })
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    let(:raw_post) { {
+      'filter' => {
+          'start_time_seconds' => {
+              'in' => ['7', '100', '4']
+          }
+      },
+      'projection' => {
+          'include' => ['id', 'start_time_seconds', 'audio_recording_id', 'creator_id']
+      }
+    }.to_json }
+    standard_request_options(:post, 'FILTER (as reader)', :ok,
+                             {
+                                 expected_json_path: 'data',
+                                 data_item_count: 0
+                             })
+  end
+
+  # sort by virtual column
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    let(:raw_post) {
+      {
+        'filter' => {
+            'start_time_seconds' => {
+                'in' => ['1', '3', '8']
+            }
+        },
+        'projection' => {
+            'include' => ['id', 'start_time_seconds', 'audio_recording_id', 'creator_id', 'order']
+        },
+        sorting: {
+            order_by: :priority,
+            direction: :asc
+        }
+      }.to_json
+    }
+
+    let!(:new_dataset_item) {
```

Comment:
this! Is good! Now apply it to this whole test suite

Reply:
fixed